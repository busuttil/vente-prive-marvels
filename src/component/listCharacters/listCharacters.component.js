import React, { Component } from "react";
import PropTypes from "prop-types";
import _ from "lodash";

import Character from "../character/character.component";
import { ListCharactersWrapper } from "./listCharacters.styles";

class ListCharacters extends Component {
  componentDidMount() {
    this.props.loadCharactersActions();
  }

  generateCharacters = () =>
    _.map(this.props.characters, character => {
      const { id, name, thumbnail } = character;
      const picture = `${thumbnail.path}.${thumbnail.extension}`;

      return <Character key={id} name={name} picture={picture} characterId={id} urls={character.urls} />;
    });

  render() {
    return (
      <ListCharactersWrapper>{this.generateCharacters()}</ListCharactersWrapper>
    );
  }
}

ListCharacters.propTypes = {
  characters: PropTypes.array.isRequired,
  loadCharactersActions: PropTypes.func.isRequired
};

export default ListCharacters;
