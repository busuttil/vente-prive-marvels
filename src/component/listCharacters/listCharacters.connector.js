import { connect } from 'react-redux';
import { loadCharacters } from '../../redux/characters/characters.actions';
import ListCharacters from './listCharacters.component';

const mapStateToProps = state => ({
  characters: state.characters.list
});

const mapDispatchToProps = dispatch => ({
  loadCharactersActions: () => dispatch(loadCharacters())
});

export default connect(mapStateToProps, mapDispatchToProps)(ListCharacters);
