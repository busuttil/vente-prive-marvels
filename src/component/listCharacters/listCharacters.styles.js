import styled from "styled-components";

export const ListCharactersWrapper = styled.div`
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
  width: 100%;
  height: 100%;
`;
