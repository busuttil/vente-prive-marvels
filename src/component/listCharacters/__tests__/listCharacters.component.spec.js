import React from 'react';
import { shallow } from 'enzyme';
import ListCharacters from "../listCharacters.component";
import { ListCharactersWrapper } from "../listCharacters.styles";

describe('ListCharacters', () => {
  let props;

  beforeEach(() => {
    props = {
      loadCharactersActions: jest.fn(),
      characters: [
        {
          name: 'Batman',
          id: 1,
          thumbnail: { path: "http://Batman", extension: "jpg" },
          urls:[]
        },
        {
          name: 'Superman',
          id: 2,
          thumbnail: { path: "http://Superman", extension: "jpg" },
          urls:[]
        }
      ],
    };
  });

  const getWrapper = () => shallow(<ListCharacters {...props} />);

  it('render Character page', () => {
    expect(getWrapper()).toMatchSnapshot();
  });

  it('render ListCharactersWrapper class', () => {
    expect(shallow(<ListCharactersWrapper />)).toMatchSnapshot();
  });
});
