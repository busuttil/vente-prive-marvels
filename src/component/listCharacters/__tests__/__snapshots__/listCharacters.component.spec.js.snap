// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListCharacters render Character page 1`] = `
<styled.div>
  <Character
    characterId={1}
    key="1"
    name="Batman"
    picture="http://Batman.jpg"
    urls={Array []}
  />
  <Character
    characterId={2}
    key="2"
    name="Superman"
    picture="http://Superman.jpg"
    urls={Array []}
  />
</styled.div>
`;

exports[`ListCharacters render ListCharactersWrapper class 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  height: 100%;
}

<div
  className="c0"
/>
`;
